src/utils.js

```diff
@@ -12,3 +13,24 @@ export function getDataHubParams(partialUrl) {
     search: parsedUrl.query,
   };
 }
+
+export function useInterval({ callback, delay, runOnStart }) {
+  const savedCallback = useRef();
+
+  useEffect(() => {
+    savedCallback.current = callback;
+  }, [callback]);
+
+  useEffect(() => {
+    function tick() {
```

Comment:
it's right define the function inside? because all times state change the function define again maybe

Reply:
That is the idea, postpouse the evaluation of _savedCallback.current_ to each tick.

This could be an alternative:

```js
const id = setInterval(() => savedCallback.current(), delay);
```

But this will not work fine:

```js
const id = setInterval(savedCallback.current, delay);
```

Reply:
That's the idea of it, I want to get the current state here.

Reply:
For more information check https://overreacted.io/making-setinterval-declarative-with-react-hooks/